File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int l = 0,r=0;
        int s = 0,mx = 0;
        int n = nums.size();
        unordered_set<int> dup;

        while(r < n){
            while(dup.find(nums[r]) != dup.end()){
                s-=nums[l];
                dup.erase(nums[l]);
                l++;
            } 
            s+=nums[r];
            dup.insert(nums[r]);
            mx = max(mx,s);
            r++;
        }     
        return mx;
    }
};

File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cs
public class Solution {
    public int MaximumUniqueSubarray(int[] nums) {
        int n = nums.Length;
        int l = 0, r =0;
        int mx = 0,s =0;

        HashSet<int> unique = new HashSet<int>();

        while(r < n){
            while(unique.Contains(nums[r])){
                s-=nums[l];
                unique.Remove(nums[l]);
                l++;
            }
            s+=nums[r];
            unique.Add(nums[r]);
            r++;
            mx = Math.Max(mx,s);
        }

        return mx;
    }
}